cluster-autoscaler/core/scaledown/unremovable/nodes.go

```diff
@@ -67,6 +67,12 @@ func (n *Nodes) Update(nodeInfos NodeInfoGetter, timestamp time.Time) {
 	n.ttls = newTTLs
 }
 
+// Contains returns true iff a given node is unremovable.
+func (n *Nodes) Contains(nodeName string) bool {
+	_, found := n.ttls[nodeName]
```

Comment:
This works for nodes added with `AddTimeout`, but not for `Add`/`AddReason`. I think checking `n.reasons` would actually deliver on the promise from the function comment. Also, since you're extending the public interface of this struct, it might be worth to add tests verifying that nodes added  via any of the `Add*` methods can then be checked for presence using `Contains`.